React/CoreModules/RCTAppearance.mm

```diff
@@ -96,6 +96,21 @@ @implementation RCTAppearance {
 
 RCT_EXPORT_MODULE(Appearance)
 
+#if TARGET_OS_OSX // [TODO(macOS GH#774)
+// We define this wrapper because [NSAppearance currentAppearance] must be called on the main thread
++ (NSAppearance *)currentAppearance {
+  if ([NSThread isMainThread]) {
+    return [NSAppearance currentAppearance];
+  } else {
+    __block NSAppearance *appearance = nil;
+    dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
this can potentially create a hang. If ui thread kicks off a background thread and waiting on result but the background thread is trying to get the nsappearance on ui thread then it is a deadlock.  what is calling appearance code in the background thread to begin with. are there things can be cached or done async?

Reply:
Was about to comment the same. There is a similar helper function in RCTUtils that is explicitly marked as unsafe https://github.com/microsoft/react-native-macos/blob/cf2105fa4915247799f3ce2f030cff5697c7aafc/React/Base/RCTUtils.m#L272

Reply:
(For my own knowledge) @harrieshin do you know if iOS has the same constraint about similar API's only being allowed to call on the main thread?

Reply:
If the UI thread is already waiting synchronously for the result of a background thread, that's already a bad thing, isn't it?

Reply:
I don't know that there are examples of the UI Thread waiting for a background thread besides at startup when all the native modules are being initialized. I think RN Core does it's best to be asynchronous and have no threads blocking each other, both to avoid hangs and to avoid deadlocks.

Reply:
In this case, I have an alternative solution that better isolates the problem.